### PR TITLE
demodel: extract ConfigHome and DataHome

### PR DESCRIPTION
**Pull request by josephholsten**
multiple locations access ~/.de for config and data use. Extract this
location into demodel so it can be self-contained and potentially
changed in future.

### CODE REVIEW

File: plugins/redmine/parseconfig.go [hunk 1]
```diff
@@ -2,18 +2,13 @@ package redmine
...
 )
 
 func parseRedminePluginConfig() {
-	u, err := user.Current()
-	if err != nil {
-		return
-	}
-
-	config, err := ioutil.ReadFile(u.HomeDir + "/.de/redmine.ini")
+	config, err := ioutil.ReadFile(demodel.ConfigHome() + "/.de/redmine.ini")
```
Comment (driusan) [inline plugins/redmine/parseconfig.go]: this seems to add an extra .de

File: demodel/charbuffer.go [hunk 2]
```diff
@@ -27,36 +27,46 @@ func (c *CharBuffer) ResetTagline() error {
...
+}
+
+// DataHome returns the directory where data files may be written
+func DataHome() string {
+	return getHomeDir() + "/.de"
+}
+
+func getSnarfSaveDir() string {
+	return DataHome() + "/snarf/"
+}
```
Comment (driusan) [inline demodel/charbuffer.go]: These three functions should probably either all include a / at the end, or all not include a / at the end, in order to make it easier to maintain in the long term

File: plugins/redmine/parseconfig.go (hunk 1 above)
Comment (josephholsten) [inline plugins/redmine/parseconfig.go]: indeed

File: demodel/charbuffer.go (hunk 2 above)
Comment (josephholsten) [inline demodel/charbuffer.go]: resolved in latest

File: plugins/redmine/parseconfig.go (hunk 1 above)
Comment (josephholsten) [inline plugins/redmine/parseconfig.go]: resolved in latest

**Pull request merged**